app/scripts/controllers/swaps.js

```diff
@@ -507,29 +515,68 @@ export default class SwapsController {
 
     let savings = null
 
+    let newQuotes = quotes
+
     if (isBest) {
+
       savings = {}
-      // Performance savings are calculated as:
-      //   valueForBestTrade - medianValueOfAllTrades
-      savings.performance = ethTradeValueOfBestQuote.minus(
-        getMedian(allEthTradeValues),
+
+      const {
+        ethValueOfQuote: ethValueOfMedianQuote,
+        ethFee: medianEthFee,
+        metaMaskFeeInEth: medianMetaMaskFee,
+      } = getMedianEthValueQuote(Object.values(quotes))
+
+      const ethValueOfMedianQuotePreMetaMaskFee = ethValueOfMedianQuote.plus(
+        medianMetaMaskFee,
+        10,
+      )
+      // Total savings are calculated as:
+      //   (overall value of best trade) - (overall value of the median trade before fees are applied)
+      savings.total = ethValueOfBestQuote.minus(
+        ethValueOfMedianQuotePreMetaMaskFee,
         10,
       )
 
-      // Performance savings are calculated as:
-      //   medianFeeOfAllTrades - feeForBestTrade
-      savings.fee = getMedian(allEthFees).minus(
+      // Fee savings are calculated as:
+      //   (fee for the median trade) - (fee for the best trade)
+      savings.fee = medianEthFee.minus(
         ethFeeForBestQuote,
         10,
       )
 
-      // Total savings are the sum of performance and fee savings
-      savings.total = savings.performance.plus(savings.fee, 10).toString(10)
-      savings.performance = savings.performance.toString(10)
+      // Performance savings are calculated by:
+      //   Subtracting the network fee savings from the total savings and adding
+      //   the metamaskFee of the best quote.
+      // savings.performance is meant to be the eth value of the difference in
+      //   token received when choosing the best quote over the median. Because
+      //   savings.total is a comparison of overall value of the best quote after
+      //   fees have been extracted to overall balue of the media quote before
+      //   fees have been extracted, savings.performance needs to adjust for the
+      //   fee extracted from the best quote. Hence the addition of the
+      //   metaMaskFeeInEthForBestQuote. This ensures that
+      //   savings.total = savings.performance + savings.fee - savings.metaMaskFee
+      //   Each of these will be represented in the UI
+      savings.performance = savings.total
```

Comment:
Wouldn't it be simpler to return the `destinationTokenAmount` (or `decimalAdjustedDestinationAmount`) directly from `getMedianEthValueQuote`, instead of having to work backwards to derive it?

Or at least that's my understanding of what's meant here - `bestQuoteDestionationAmount - medianQuoteDestinationAmount - metaMaskFee`